quantized_mesh_encoder/encode.py

```diff
@@ -3,14 +3,14 @@
 import numpy as np
 
 from .bounding_sphere import bounding_sphere
-from .constants import HEADER, NP_STRUCT_TYPES, VERTEX_DATA
+from .constants import HEADER, NP_STRUCT_TYPES, VERTEX_DATA, EXTENSION_HEADER
 from .ecef import to_ecef
 from .occlusion import occlusion_point
 from .util import zig_zag_encode
 from .util_cy import encode_indices
+from .normals import compute_vertex_normals, oct_encode
 
-
-def encode(f, positions, indices, bounds=None, sphere_method=None):
+def encode(f, positions, indices, bounds=None, sphere_method=None, generate_normals=False):
```

Comment:
I think a bit more thought is warranted about what the public API should be. If there are more extensions in the future, should there be a new parameter for each one? Or should there be some sort of `extensions` param that designates which extensions should be computed and included.

Reply:
Seems the three supported extensions in the spec are: `OctEncodedVertexNormals`, `WaterMask`, and `Metadata`.

So if i follow you correctly have encode accept something like:
```python
extensions={"octEncodedVertexNormals": True}
```

which could easily be extended to include the others if/when they are added

Reply:
Yes, that's the general idea, or something like

```py
from enum import IntEnum

class QuantizedMeshExtensions(IntEnum):
	VERTEX_NORMALS = 1
	WATER_MASK = 2
	METADATA = 4

encode(extensions=[QuantizedMeshExtensions.VERTEX_NORMALS, QuantizedMeshExtensions.WATER_MASK])
```

The issue there is that some extensions will need extra data. For example, vertex normals will be computed from the input data, but computing a water mask is definitely out of scope of this package, so there would need to be an argument to accept water mask data.

I've recently gotten hooked on using dataclasses, so a possibility here is for each input extension to be a dataclass.

```py
@dataclass
class ExtensionBase:
	extension_id: str

@dataclass
class VertexNormalsExtension(ExtensionBase):
	extension_id: str = QuantizedMeshExtensions.VERTEX_NORMALS

@dataclass
class WaterMaskExtension(ExtensionBase):
	extension_id: str = QuantizedMeshExtensions.WATER_MASK
	data: Union[np.array, int]

vertex_ext = VertexNormalsExtension()
watermask_ext = WaterMaskExtension(data=255)
# or
watermask_ext = WaterMaskExtension(data=np.zeros(shape=(256, 256), dtype=np.uint8))

encode(extensions=[vertex_ext, watermask_ext])
```

Then inside `encode` you can check that all arguments to `extensions` are instances of `ExtensionBase` and are all unique extensions